tests/testthat/test-caretList.R

```diff
@@ -144,6 +144,31 @@ test_that("caretList predictions", {
   expect_error(expect_warning(predict(models)))
 })
 
+test_that("as.caretList returns a caretList object", {
+
+  expect_warning({
+    modelList <- caretList(Sepal.Length ~ Sepal.Width,
+                           head(iris, 50),
+                           methodList=c("glm", "lm"))
+  })
+
+  class(modelList) <- "list"
+
+  expect_is(as.caretList(modelList), "caretList")
+})
+
+test_that("[.caretList test caretList indexing", {
+
+  expect_warning({
+    modelList <- caretList(Sepal.Length ~ Sepal.Width,
+                           head(iris, 50),
+                           methodList=c("glm", "lm"))
```

Comment:
For this test, lets use 3 models, something like `methodList=c("glm", "lm", "knn")` and then subset the list with something like `modelList <- modelList[1:2]`.

Then we can check `expect_is(as.caretList(modelList), "caretList")` and make sure the subsetted object is still a caretList